src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer/StrategyFactory.php

```diff
@@ -77,12 +77,12 @@ class StrategyFactory
     /**
      * List of all valid properties.
      *
-     * @var array<string>
+     * @var array<string, class-string<CodeRankStrategyI>>
      */
     private $validStrategies = array(
-        self::STRATEGY_INHERITANCE,
-        self::STRATEGY_METHOD,
-        self::STRATEGY_PROPERTY
+        self::STRATEGY_INHERITANCE => 'PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\InheritanceStrategy',
+        self::STRATEGY_METHOD => 'PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\MethodStrategy',
+        self::STRATEGY_PROPERTY => 'PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\PropertyStrategy',
```

Comment:
Is there a reason why we can't use the class property (plus importing the classes to simplify the FQCN)?

```suggestion
        self::STRATEGY_INHERITANCE => PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\InheritanceStrategy::class,
        self::STRATEGY_METHOD => PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\MethodStrategy::class,
        self::STRATEGY_PROPERTY => PDepend\\Metrics\\Analyzer\\CodeRankAnalyzer\\PropertyStrategy::class,
```

Reply:
I'm thinking it's better to do that as a big update for the whole repo later. It's one of the thinks that php-cs-fixer can solve for us, so I didn't want to spend to much time one it, or have mixed styles.